sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcQuerySuite.scala

```diff
@@ -270,10 +270,16 @@ abstract class OrcQueryTest extends OrcTest {
   test("appending") {
     val data = (0 until 10).map(i => (i, i.toString))
     spark.createDataFrame(data).toDF("c1", "c2").createOrReplaceTempView("tmp")
-    withOrcTable(data, "t") {
-      sql("INSERT INTO TABLE t SELECT * FROM tmp")
-      checkAnswer(spark.table("t"), (data ++ data).map(Row.fromTuple))
+
+    withOrcFile(data) { file =>
```

Comment:
why can't we use `withOrcTable` here?

Reply:
Previously, `withOrcTable` runs only vectorized case. After this, `withOrcTable` runs vectorized and non-vectorized cases. Because the test closure has side effect that modifies table `t`. So in the second run, the table `t` is changed and can't match the expected answer.